packages/debug/src/browser/run-in-terminal.ts

```diff
@@ -0,0 +1,145 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { injectable } from 'inversify';
+import { createShellCommandLine, BashQuotingFunctions, PowershellQuotingFunctions, CmdQuotingFunctions, ShellQuoting, ShellQuotedString } from '@theia/process/lib/common/shell-quoting';
+import { TerminalProcessInfo } from '@theia/terminal/lib/common/base-terminal-protocol';
+
+export interface CommandLineOptions {
+    cwd: string, args: string[], env?: {
+        [key: string]: string | null
+    }
+}
+
+@injectable()
+export class RunInTerminalService {
```

Comment:
EIther prefix with `Debug` or move to `@theia/terminal`, please see https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#unique-names

Reply:
RunInTerminal felt like a debug-specific concept, but you are right. Will move.